libnavigation-core/api/current.txt

```diff
@@ -830,6 +834,27 @@ package com.mapbox.navigation.core.telemetry.events {
     method public com.mapbox.navigation.core.telemetry.events.FeedbackMetadata get();
   }
 
+  public final class UserFeedback {
```

Comment:
should the API be internal? Did we agree we want to expose it?

Reply:
It is necessary for Copilot, which is a separate library, so it has to be public. We can move it to internal package, but I'm afraid we might accidentally break compatibility with Copilot if we change something.

Reply:
let's move to the internal package first, then we have a chance to change the API or decline this approach at all. I'm not sure if it will be easy to support when we migrate to the Core Telemetry or move Telemetry to Native. It's easier to expose than to hide

Reply:
What do you think of adding `ExperimentalPreviewMapboxNavigationAPI` annotation?

Reply:
> What do you think of adding ExperimentalPreviewMapboxNavigationAPI annotation?

That should be the way to do it if we're not confident about the API surface yet, for both the `UserFeedback` and corresponding observer methods. We can't make data type internal or experimental while leaving the observer registration stable, that would be "leaking" the implementation. They need to move in unison.

Reply:
If we put everything under the internal package, including registration/unregistration we should be good. For instance, it might look like 
```kotlin
package com.mapbox.navigation.[package].internal

fun registerObserverX(mapboxNavigation: MapboxNavigation, observerX: ObserverX){
  mapboxNavigation.registerObserverX(observerX)
}

```

Reply:
@RingerJK why not use experimental annotation instead? We won't have to duplicate methods this way. Also this API will be present in current.txt files, so we will be able to easily track its compatibility violations.

Reply:
I don't think this API will be removed, because Copilot needs it. Let's wait for @Guardiola31337's opinion.

Reply:
I'm fine with having these APIs under the `internal` package. Actually, Copilot is an internal library and we should be good accommodating to any changes 👍

Reply:
In other words, I'd rather keeping it `internal` for now if possible.